### PR TITLE
feat(mark-support-level): IPNS whitelist #3259

### PR DESCRIPTION
**Pull request by tx0c**
in the DB table, but actual work will be done in a cron service in lambda

### CODE REVIEW

File: src/mutations/system/setUsersIPNSSupportLevel.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import createDebug from 'debug'
+
+import { UserInputError } from 'common/errors'
+import { fromGlobalId } from 'common/utils'
+import { MutationToSetUsersIPNSSupportLevelResolver } from 'definitions'
+
+const debugLog = createDebug('users-ipns')
```
Comment (robertu7) [inline src/mutations/system/setUsersIPNSSupportLevel.ts:7]: @gary02 related to your ticket (#3249 ) ?

File: src/connectors/atomService.ts [hunk 2]
```diff
@@ -227,8 +232,30 @@ export class AtomService extends DataSource {
    *
    * A Prisma like method for updating many records.
    */
-  updateMany = async ({ table, where, data, columns = '*' }: UpdateInput) =>
-    this.knex.where(where).update(data).into(table).returning(columns)
+  updateMany = async ({
```
Comment (robertu7) [inline src/connectors/atomService.ts:235]: will merge back to https://github.com/thematters/matters-server/pull/3272

File: src/mutations/system/setUsersIPNSSupportLevel.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  if (!ids || ids.length === 0) {
+    throw new UserInputError('"ids" is required')
+  }
+
+  const userIds = ids.map((id) => fromGlobalId(id).id)
+
+  const updated = await atomService.updateMany({
+    table: 'user_ipns_keys',
+    whereIn: ['user_id', userIds],
+    data: { supportLevel },
```
Comment (robertu7) [inline src/mutations/system/setUsersIPNSSupportLevel.ts:23]: we can reset the record via passing a `supportLevel: undefined`?

**Review by robertu7 — APPROVED** [review]

File: src/mutations/system/setUsersIPNSSupportLevel.ts (hunk 3 above)
Comment (tx0c) [reply src/mutations/system/setUsersIPNSSupportLevel.ts:23]: can reset to `null`

**Pull request closed without merging**